src/functions/getDelegatedBooking/framework/__tests__/handler.spec.ts

```diff
@@ -0,0 +1,137 @@
+    dummyApigwEvent.requestContext.authorizer = { examinerRole: 'DLG' };
+    dummyContext = lambdaTestUtils.mockContextCreator(() => null);
+    spyOn(FindDelegatedBooking, 'findDelegatedBooking').and.callFake(moqFindDelegatedBooking.object);
+  });
+
+  describe('given the FindJournal returns a journal', () => {
+    it('should return a successful response with the journal', async () => {
+      moqFindDelegatedBooking.setup(x => x(It.isAny())).returns(() => Promise.resolve(booking));
+      createResponseSpy.and.returnValue({ statusCode: 200 });
+
+      const resp = await handler(dummyApigwEvent, dummyContext);
+
+      expect(resp.statusCode).toBe(200);
+      expect(createResponse.default).toHaveBeenCalledWith(booking);
+    });
+  });
+
+  describe('given FindJournal throws a JournalNotFound error', () => {
+    it('should return HTTP 404 NOT_FOUND', async () => {
+      moqFindDelegatedBooking.setup(x => x(It.isAny())).throws(new DelegatedBookingNotFoundError());
+      createResponseSpy.and.returnValue({ statusCode: 404 });
+
+      const resp = await handler(dummyApigwEvent, dummyContext);
+
+      expect(resp.statusCode).toBe(404);
+      expect(createResponse.default).toHaveBeenCalledWith({}, 404);
+    });
+  });
+
+  describe('given the FindJournal throws', () => {
+    it('should respond with internal server error', async () => {
+      moqFindDelegatedBooking.setup(x => x(It.isAny())).throws(new Error('Unable to retrieve delegated booking'));
+      createResponseSpy.and.returnValue({ statusCode: 500 });
+
+      const resp = await handler(dummyApigwEvent, dummyContext);
+
+      expect(resp.statusCode).toBe(500);
+      expect(createResponse.default).toHaveBeenCalledWith('Unable to retrieve delegated booking', 500);
+    });
+  });
+
+  describe('given there is no data but successful request', () => {
+    it('should indicate a successful request with no data', async () => {
+      moqFindDelegatedBooking.setup(x => x(It.isAny())).returns(() => Promise.resolve(null));
+      createResponseSpy.and.returnValue({ statusCode: 204 });
+
+      const resp = await handler(dummyApigwEvent, dummyContext);
+
+      expect(resp.statusCode).toBe(204);
+      expect(createResponse.default).toHaveBeenCalledWith({}, 204);
+    });
+  });
+  describe('given there is no applicationReference provided', () => {
+    it('should indicate a bad request', async () => {
+      dummyApigwEvent.pathParameters = {};
+      createResponseSpy.and.returnValue({ statusCode: 400 });
+
+      const resp = await handler(dummyApigwEvent, dummyContext);
+
+      expect(resp.statusCode).toBe(400);
+      expect(createResponse.default).toHaveBeenCalledWith('No applicationReference provided', 400);
+    });
+  });
+  describe('given there app ref is in wrong format', () => {
+    it('should indicate a bad request', async () => {
+      dummyApigwEvent.pathParameters = { applicationReference: '123' };
+      createResponseSpy.and.returnValue({ statusCode: 400 });
```

Comment:
if we're forcing createResponse to return a 400 then we're not testing the handler's ability to validate app refs